webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,312 @@
+import React from 'react';
+import './ticketStyle.scss';
+
+import {Dispatch} from 'redux';
+
+import {Instance} from 'types/model';
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    isLoaded?: boolean;
+    ticketDetails?: TicketDetails;
+    defaultUserInstanceID?: string;
+    connectedInstances: Instance[];
+    getIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+        error?: any;
+    }>;
+    setTicket?: (ticketDetails: {}) => void;
+}
+
+export type State = {
+    href: string;
+    isLoaded: boolean;
+    ticketId: string;
+    ticketDetails?: TicketDetails;
+};
+
+const isAssigned = ' is assigned';
+const unAssigned = 'Unassigned';
+const jiraTicketTitleMaxLength = 80;
+
+enum myStatus {
+    INDETERMINATE = 'indeterminate',
+    DONE = 'done',
+}
+
+const myStatusClasses: Record<string, string> = {
+    [myStatus.INDETERMINATE]: ' ticket-status--indeterminate',
+    [myStatus.DONE]: ' ticket-status--done',
+};
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    truncateString(str: string, num: number) {
+        if (num > str.length) {
+            return str;
+        }
+        return `${str.substring(0, num)}...`;
+    }
+
+    constructor(Props: Props) {
+        super(Props);
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+        }
+
+        if (!ticketID && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+        }
+
+        this.state = {
+            href: this.props.href,
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    getIssueKey = () => {
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+        }
+
+        if (!ticketID && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+        }
+
+        return ticketID;
+    }
+
+    init() {
+        let instanceID = this.props.defaultUserInstanceID || '';
+
+        if (this.props.connectedInstances.length === 1) {
+            instanceID = this.props.connectedInstances[0].instance_id;
+        }
+
+        const ticketID = this.getIssueKey();
+        if (ticketID) {
+            this.props.getIssueByKey(ticketID, instanceID);
+        }
+    }
+
+    componentDidMount() {
+        const {connected, ticketDetails, isLoaded: isPropsLoaded} = this.props;
+        const {ticketId, isLoaded: isStateLoaded} = this.state;
+        if (connected && !isStateLoaded && ((ticketDetails && ticketDetails.ticketId) !== ticketId || !isPropsLoaded)) {
```

Comment:
this line and the condition in 101 is almost the same, it's better to move it to a method/function and use it twice in these lines, also easier to test.